pkg/client/options.go

```diff
@@ -0,0 +1,65 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"github.com/spf13/pflag"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+type ClientOpts struct {
+	which        string
+	description  string
+	loadingRules *clientcmd.ClientConfigLoadingRules
+	overrides    clientcmd.ConfigOverrides
+}
+
+func NewClientOpts(which, description string) *ClientOpts {
+	return &ClientOpts{
+		which:        which,
+		description:  description,
+		loadingRules: clientcmd.NewDefaultClientConfigLoadingRules(),
+		overrides:    clientcmd.ConfigOverrides{},
+	}
+}
+
+func (opts *ClientOpts) OverrideCurrentContext(currCtx string) {
```

Comment:
If this returned the receiver then these calls could be chained

Reply:
Done in the latest force-push. Thanks!